policy-test/tests/accepted_http_route_status.rs

```diff
@@ -0,0 +1,421 @@
+            .await
+            .parents
+            .get(0)
+            .expect("must have at least one parent status")
+            .conditions
+            .iter()
+            .find(|cond| cond.type_ == "Accepted")
+            .expect("must have at least one 'Accepted' condition")
+            .clone();
+        assert_eq!(cond.status, "False");
+        assert_eq!(cond.reason, "NoMatchingParent");
+
+        // Save create_timestamp to assert status has been updated.
+        let create_timestamp = &cond.last_transition_time;
+
+        // Create the 'Server' that route references and expect it to be picked up
+        // by the index. Consequently, route will have its status reconciled.
+        let _server = {
+            let server = k8s::policy::Server {
+                metadata: k8s::ObjectMeta {
+                    namespace: Some(ns.to_string()),
+                    name: Some("test-reconcile-create-server".to_string()),
+                    ..Default::default()
+                },
+                spec: k8s::policy::ServerSpec {
+                    pod_selector: k8s::labels::Selector::from_iter(Some((
+                        "app",
+                        "test-reconcile-create-server",
+                    ))),
+                    port: k8s::policy::server::Port::Name("http".to_string()),
+                    proxy_protocol: Some(k8s::policy::server::ProxyProtocol::Http1),
+                },
+            };
+            create(&client, server).await
+        };
+
+        // HTTPRoute may not be patched instantly, wrap with a timeout and loop
+        // until create_timestamp and observed_timestamp are different.
+        let cond = tokio::time::timeout(tokio::time::Duration::from_secs(60), async move {
+            loop {
+                let cond = await_route_status(&client, &ns, "test-reconcile-create-route")
+                    .await
+                    .parents
+                    .get(0)
+                    .expect("must have at least one parent status")
+                    .conditions
+                    .iter()
+                    .find(|cond| cond.type_ == "Accepted")
+                    .expect("must have at least one 'Accepted' condition")
+                    .clone();
+
+                // Observe condition's current timestamp. If it differs from the
+                // previously recorded timestamp, then it means the underlying
+                // condition has been updated so we can check the message and
+                // status.
+                if &cond.last_transition_time > create_timestamp {
+                    return cond;
+                }
+            }
+        })
+        .await
+        .expect("Timed-out waiting for HTTPRoute status update");
+        assert_eq!(cond.status, "True");
+        assert_eq!(cond.reason, "Accepted");
+    })
+    .await;
+}
+
+#[tokio::test(flavor = "current_thread")]
+async fn inbound_accepted_reconcile_delete_event() {
```

Comment:
That was my original intent, buut it was simpler to create a new test. I tried to put both of them in the same test and I ended up reverting the changes. Some of the reasons why:

* If we use them both, we have to do the whole clone and dance ritual. Since we move client & namespace into the async blocks we have to clone them prior to it. If we have them together, we have two async blocks (since we need to wait for the status to be updated twice). Not the end of the world but it doesn't add to readability
* I thought the failure point might be a bit more straightforward if the functions test each other's inverse (e.g we know we failed when we went from parent -> no parent).
* Since they execute concurrently we have less waiting time (not that 1m worst-case time is _bad_, we already run the multicluster tests lol)

Most of these are trivial, I think my biggest motivation was that less things happen so imo it's a bit simpler to come back to. I did change the name of the tests though since maybe the `create/delete` doesn't really reflect the behaviour. I'm also okay with making this change, will default to your wisdom here :D